enterprise/mqtt/src/main/java/io/crate/mqtt/operations/MqttIngestService.java

```diff
@@ -19,47 +19,108 @@
-    private static final String STMT_NAME = "I";
-    private static final String STMT = "INSERT INTO raw (" +
-                                       "\"client_id\", \"packet_id\", \"topic\", \"ts\", \"payload\"" +
-                                       ") VALUES (" +
-                                       "?, ?, ?, CURRENT_TIMESTAMP, ?" +
-                                       ")";
-    private static final List<DataType> DATA_TYPES = Arrays.asList(DataTypes.STRING, DataTypes.INTEGER, DataTypes.STRING, DataTypes.OBJECT);
     private static final Logger LOGGER = Loggers.getLogger(MqttIngestService.class);
+    static final String SOURCE_IDENT = "mqtt";
+    private static final Map<String, Integer> MQTT_FIELDS_ORDER = ImmutableMap.of("client_id", 0,
+        "packet_id", 1,
+        "topic", 2,
+        "ts", 3,
+        "payload", 4);
+    private static final List<DataType> DATA_TYPES = Arrays.asList(DataTypes.STRING, DataTypes.INTEGER, DataTypes.STRING, DataTypes.OBJECT);
+    private static final String STATEMENT_NAME = "ingestMqtt";
 
-    private final Session session;
+    private final ExpressionAnalyzer expressionAnalyzer;
+    private final InputFactory inputFactory;
+    private final SQLOperations sqlOperations;
+    private final IngestionService ingestionService;
+    private final User crateUser;
+    private final AtomicReference<Set<Tuple<Predicate<Row>, IngestRule>>> predicateAndIngestRulesReference =
+        new AtomicReference<>(new HashSet<>());
+    private final ExpressionAnalysisContext expressionAnalysisContext;
+    private boolean isInitialized;
 
-    public MqttIngestService(SQLOperations sqlOperations) {
-        session = sqlOperations.createSession(DEFAULT_SCHEMA, null, Option.NONE, 1);
-        session.parse(STMT_NAME, STMT, DATA_TYPES);
+    public MqttIngestService(Functions functions,
+                             SQLOperations sqlOperations,
+                             UserManager userManager,
+                             IngestionService ingestionService) {
+        this.sqlOperations = sqlOperations;
+        this.inputFactory = new InputFactory(functions);
+        this.expressionAnalysisContext = new ExpressionAnalysisContext();
+        FieldProvider<Symbol> mqttSourceFieldsProvider = new FieldProvider<Symbol>() {
+
+            @Override
+            public Symbol resolveField(QualifiedName qualifiedName, Operation operation) {
+                return resolveField(qualifiedName, null, operation);
+            }
+
+            @Override
+            public Symbol resolveField(QualifiedName qualifiedName, @Nullable List<String> path, Operation operation) {
+                return new InputColumn(MQTT_FIELDS_ORDER.get(qualifiedName.toString()));
+            }
+        };
+        this.expressionAnalyzer = new ExpressionAnalyzer(functions, null, null, mqttSourceFieldsProvider, null);
+        this.ingestionService = ingestionService;
+        this.crateUser = userManager.findUser("crate");
```

Comment:
Either change the dependency to just `UserLookup` or maybe even depend on a user directly.